contracts/ERC-721/ERC721Pool.sol

```diff
@@ -54,7 +79,26 @@ contract ERC721Pool is IERC721Pool, ERC20Wnft {
     }
 
     /// @inheritdoc IERC721Pool
-    function withdraw(uint256[] calldata ids) public override {
+    function rescueLastNFT(address to) external override onlyFactory {
+        if (holdings.length() != 1) {
+            revert ERC721Pool__MustContainExactlyOneNFT();
+        }
+        uint256 lastNFT = holdings.at(0);
+        require(holdings.remove(lastNFT));
+        IERC721(asset).transferFrom(address(this), to, lastNFT);
+        poolFrozen = true;
+        emit PoolFrozen();
```

Comment:
@surbhiaudichya I think it should emit `RescueLastNFT` instead with the NFT ID

Reply:
Thanks @scorpion9979, After considering your suggestion, I agree that it makes sense to emit the RescueLastNFT event within the ERC721Pool contract, including the NFT ID.  
I will make the necessary changes to emit the RescueLastNFT event in the ERC721Pool contract and remove the event from the ERC721PoolFactory contract. and remove the PoolFrozen event from ERC721Pool contract.